--- a/spa/plugins/alsa/90-pipewire-alsa.rules
+++ b/spa/plugins/alsa/90-pipewire-alsa.rules
@@ -207,6 +207,9 @@
 # hikvision
 ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="4f36", ENV{ACP_PROFILE_SET}="hikvision-usb-audio.conf"
 
+# Lenovo M90H G1s
+ATTRS{idVendor}=="17aa", ATTRS{idProduct}=="3501", ENV{ACP_PROFILE_SET}="lenovo-m90h-g1s-usb-audio.conf"
+
 GOTO="pipewire_end"
 
 LABEL="pipewire_check_pci"
--- /dev/null
+++ b/spa/plugins/alsa/mixer/profile-sets/lenovo-m90h-g1s-usb-audio.conf
@@ -0,0 +1,30 @@
+[General]
+auto-profiles = yes
+
+[Mapping analog-stereo-speaker-output]
+description = Analog Stereo Speaker
+device-strings = hw:%f,0
+channel-map = left,right
+direction = output
+paths-output = analog-output-speaker
+
+[Mapping analog-stereo-headphone-output]
+description = Analog Stereo Headphones
+device-strings = hw:%f,1
+channel-map = left,right
+direction = output
+paths-output = analog-output-headphones
+
+[Mapping analog-stereo-line-input]
+description = Analog Stereo Line Input
+device-strings = hw:%f,1
+channel-map = left,right
+direction = input
+paths-input = analog-input-linein
+
+[Mapping analog-stereo-mic-input]
+description = Analog Stereo Input
+device-strings = hw:%f,2
+channel-map = left,right
+direction = input
+paths-input = analog-input-mic
